libs/output-mapping/src/Storage/TableStructureValidator.php

```diff
@@ -0,0 +1,186 @@
+     */
+    private function validateTypedTable(array $table, array $schemaColumns, string $bucketBackend): void
+    {
+        $validationErrors = [];
+        foreach ($schemaColumns as $schemaColumn) {
+            if (!$schemaColumn->getDataType()) {
+                continue;
+            }
+            $filteresTableColumns = array_filter(
+                $table['definition']['columns'],
+                fn($column) => $column['name'] === ColumnNameSanitizer::sanitize($schemaColumn->getName()),
+            );
+
+            $tableColumn = current($filteresTableColumns);
+            try {
+                $schemaColumnType = $schemaColumn->getDataType()->getTypeName($bucketBackend);
+                $tableColumnType = $tableColumn['definition']['type'];
+            } catch (InvalidOutputException) {
+                $schemaColumnType = $schemaColumn->getDataType()->getBaseType();
+                $tableColumnType = $tableColumn['basetype'];
+            }
+
+            if (strtolower($schemaColumnType) !== strtolower($tableColumnType)) {
+                $validationErrors[] = sprintf(
+                    'Table "%s" column "%s" has different type than the schema.'.
+                    ' Table type: "%s", schema type: "%s".',
+                    $table['id'],
+                    $schemaColumn->getName(),
+                    $tableColumnType,
+                    $schemaColumnType,
+                );
+            }
+
+            try {
+                $schemaColumnLength = $schemaColumn->getDataType()->getLength($bucketBackend);
+            } catch (InvalidOutputException) {
+                $schemaColumnLength = $schemaColumn->getDataType()->getBaseLength();
+            }
+
+            if ($schemaColumnLength && $schemaColumnLength !== $tableColumn['definition']['length']) {
+                $validationErrors[] = sprintf(
+                    'Table "%s" column "%s" has different length than the schema.'.
+                    ' Table length: "%s", schema length: "%s".',
+                    $table['id'],
+                    $schemaColumn->getName(),
+                    $tableColumn['definition']['length'],
+                    $schemaColumnLength,
+                );
+            }
+
+            if ($schemaColumn->isNullable() !== $tableColumn['definition']['nullable']) {
+                $validationErrors[] = sprintf(
+                    'Table "%s" column "%s" has different nullable than the schema.'.
+                    ' Table nullable: "%s", schema nullable: "%s".',
+                    $table['id'],
+                    $schemaColumn->getName(),
+                    $tableColumn['definition']['nullable'] ? 'true' : 'false',
+                    $schemaColumn->isNullable() ? 'true' : 'false',
+                );
+            }
+        }
+
+        if ($validationErrors) {
+            throw new InvalidTableStructureException(implode(' ', $validationErrors));
```

Comment:
Ideálně by tu měl být test, že to nesletí na první chybě, ale vrátí všechny.
Mapping s 2 chybama =>  exception se dvěma message